flutter-studio/src/io/flutter/android/GradleDependencyFetcher.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.android;
+
+import com.android.annotations.VisibleForTesting;
+import com.android.prefs.AndroidLocation;
+import com.android.tools.idea.gradle.project.common.GradleInitScripts;
+import com.android.tools.idea.gradle.util.EmbeddedDistributionPaths;
+import com.android.utils.SdkUtils;
+import com.google.common.collect.Lists;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.CapturingProcessHandler;
+import com.intellij.execution.process.ProcessOutput;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.containers.hash.HashMap;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+import java.util.Map;
+
+import static com.android.SdkConstants.FN_GRADLE_WRAPPER_UNIX;
+import static com.android.SdkConstants.FN_GRADLE_WRAPPER_WIN;
+
+public class GradleDependencyFetcher {
+  private static final Logger LOG = Logger.getInstance(GradleDependencyFetcher.class);
+  private static final String DEP_COMMENT = " - ";
+  private static final String DEP_LIST_ITEM = "+---";
+  private static final String DEP_LIST_END = "\\---";
+  private static final String DEP_LIST_TERMINATOR = "--- ";
+
+  private final Project myProject;
+  private final Map<String, List<String>> myDependencies;
+  private String myType;
+  private String mySection;
+  private List<String> mySectionDependencies;
+
+  public GradleDependencyFetcher(@NotNull Project project) {
+    myProject = project;
+    myDependencies = new HashMap<>(15);
+  }
+
+  @NotNull
+  public Map<String, List<String>> getDependencies() {
+    return myDependencies;
+  }
+
+  public void run() {
```

Comment:
When this is used, I assume the result will either be available synchronously, or asynchronously via a Future?

Reply:
Well, run() is synchronous. But I expect we'll have something driving the process that is async and also takes care of the progress UI.